quarkchain/cluster/shard_state.py

```diff
@@ -1668,10 +1667,13 @@ def run_tx(gas):
             return None
         return hi
 
-    def gas_price(self) -> Optional[int]:
+    def gas_price(self, token_id: int) -> Optional[int]:
         curr_head = self.header_tip.get_hash()
-        if curr_head == self.gas_price_suggestion_oracle.last_head:
-            return self.gas_price_suggestion_oracle.last_price
+        if (
```

Comment:
imagine the following:

1. query QKC gas price at block 10 -> last_head => 10, last_price => {QKC: 123}
2. query QI gas price at block 11 -> last_head => 11, last_price => {QKC: 123, QI: 456}
3. query QKC gas price again at block 11 -> directly return 123, which is wrong because it's for block 10